### ISSUE TITLE
dyld: Library not loaded ... image not found

### ISSUE THREAD

**Issue by michaeleisel** [CocoaPods/CocoaPods#9156]
* [X] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Create Podfile for empty MacOS CLI tool with:
```
target 'podNew' do
  use_frameworks!

  pod 'RxSwift'
  # Pods for podNew

end
```

Then run `pod install` and build

## What did you expect to happen?

A successful build

## What happened instead?

```
dyld: Library not loaded: @rpath/RxSwift.framework/Versions/A/RxSwift
  Referenced from: /Users/michaeleisel/Documents/Projects/podNew/Build/Products/Debug/podNew
  Reason: image not found
```

### Stack

```
   CocoaPods : 1.7.5
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15 (19A536g)
       Xcode : 10.3 (10G8)
         Git : git version 2.21.0
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : artsy - https://github.com/artsy/Specs.git @ ef1735b844309562ba59a1ef257a3148958fe4e9
               master - https://github.com/CocoaPods/Specs.git @ e32f4aaa8f6ad641a62cf88b3b54b521055331d0
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate                 : 1.0.4
cocoapods-disable-podfile-validations : 0.1.1
cocoapods-generate                    : 1.5.0
cocoapods-plugins                     : 1.0.0
cocoapods-search                      : 1.0.0
cocoapods-stats                       : 1.1.0
cocoapods-trunk                       : 1.3.1
cocoapods-try                         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'podNew' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  pod 'RxSwift'
  # Pods for podNew

end
```

## Project that demonstrates the issue

It's just an empty project, created and built with Xcode 10.

**Comment by michaeleisel** [comment]
It seems that there are other issues like this one related to CLI tools, e.g. https://github.com/CocoaPods/CocoaPods/issues/3709, but the steps from that guide are outdated (e.g. `link_with`), and there didn't seem to be a really clear answer anyways.

**Comment by dnkoutso** [comment]
We do not support macOS CLI tools in CocoaPods IIRC.

**Comment by michaeleisel** [comment]
I see, which types of Mac apps are supported? And could there be some sort of warning when the user runs `pod install`, telling them that it isn't supported? I think that other users would be surprised by that as well

**Comment by dnkoutso** [comment]
@michaeleisel actually I think @amorde implemented a warning here https://github.com/CocoaPods/CocoaPods/issues/6493! It has shipped with 1.8.0.beta.1

**Comment by dnkoutso** [comment]
We can probably further enhance it with the list of supported targets. But I would say Applications and Frameworks / Libraries are supported for macOS.

**Comment by michaeleisel** [comment]
OK thank you

**Comment by amorde** [comment]
Yup we added a warning when using dynamic frameworks with CLI targets, since there’s no way to embed the framework into the binary.

CLI targets should work though, as long as you’re using static linking or are able to package the dynamic frameworks with the binary somehow

**Comment by kokominaj** [comment]
I get this error when I install when I do `brew install awslogs` and then I get this error, after that to fix this I have to do, `brew reinstall https://github.com/tebelorg/Tump/releases/download/v1.0.0/openssl.rb` and then it gets fixed but then my awslogs brakes so I do `brew reinstall awslogs` again and then my awslogs gets fixed but this is broken now, so I have to reinstall openssl again , but then I get same error again, due to this error I am stuck in infinite loop, as awslogs require a higher version of openssl.
